src/java.desktop/macosx/classes/sun/lwawt/macosx/CPrinterJob.java

```diff
@@ -860,13 +895,18 @@ private Object[] getPageformatPrintablePeekgraphics(final int pageIndex) {
                 if (pageFormat != null) {
                     Printable printable = pageable.getPrintable(pageIndex);
                     if (printable != null) {
+                        double scaleX = getXRes() / USER_SPACE_DPI;
+                        double scaleY = getYRes() / USER_SPACE_DPI;
+                        PageFormat scaledPageFormat = scalePageFormat(pageFormat, scaleX, scaleY);
                         BufferedImage bimg =
                               new BufferedImage(
-                                  (int)Math.round(pageFormat.getWidth()),
-                                  (int)Math.round(pageFormat.getHeight()),
+                                  (int)Math.round(scaledPageFormat.getWidth()),
+                                  (int)Math.round(scaledPageFormat.getHeight()),
```

Comment:
Wow, that is going to be one big BufferedImage. This doesn't seem right or necessary for a PeekGraphics.

Look at RasterPrinterGraphics. It uses a 1x1 image (!) It just sets things up so that when the app asks for the graphics config it gets the page dimensions back.